File: backend/server/models/category.go
package models

type Category struct {
	// The ID of the category.
	//
	// Will be ignored when writing.
	Id          int    `json:"id"`
	Name        string `json:"name" binding:"required,max=45"`
	Description string `json:"description" binding:"max=255"`
	Internal    bool   `json:"internal" binding:"required"`
}

type CategoryAddBody struct {
	Categories []Category `json:"categories" binding:"required,dive"`
}

File: expensetrack/lib/models/category.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'category.freezed.dart';
part 'category.g.dart';

@freezed
class Category with _$Category {
  factory Category
({
    /// The ID of the category.
    /// 
    /// Will be ignored when writing.
    required int? id,
    
    /// The name of the category.
    /// 
    /// Max 40 characters.
    required String name,
    
    /// The category description.
    /// 
    /// Max 255 characters.
    required String? description,
		
    /// Whether the transactions in this category are internal.
    /// 
    /// Internal transactions will not be included in certain statistics.
    required bool internal,
  }) = _Category
;

factory Category.fromJson(Map<String, dynamic> json) => 
_$CategoryFromJson(json);
}